### DISCUSSION TITLE
Better hints

### DISCUSSION THREAD

**Discussion by chaitanyabsprip** [ggandor/leap.nvim#90] [Q&A] [answered]
I am using a custom keyboard layout and want to change the default hints. I am confused as to what `safe_lables` mean. Also, how is the priority of labels determined. 

Before leap, I used to use hop, and I had the following string set as hints to work with hop's algorithm. `'nepoihufxbzyqjkcvlmwdasgrt' `

**Comment by ggandor** [comment, marked as answer]
`:help leap.opts.safe_labels` (see also [smart autojump](https://github.com/ggandor/leap.nvim/tree/e0145906c5f004b23eb6ec876fad55ffd3382ec9#smart-autojump)). If there's anything else to be clarified, feel free to ask.

> Also, how is the priority of labels determined.

They are assigned in the same order as they appear in the table, simple as that. If you have `labels = {'a', 'b', 'c'}`, then the first target will get `a`, and so on... (for `s`/`S`, the targets are ordered by distance in the buffer, towards the end/beginning of the file, in multi-window mode, the beeline distance from the cursor).

**Comment by jgregoire** [comment]
This is a great feature. I use Norman layout with a ton of remapping. It was easy to make leap play nice.
